src/Microsoft.Azure.WebJobs.Extensions.EventHubs/Listeners/EventHubScaleMonitor.cs

```
@@ -0,0 +1,395 @@
+            {
+                _logger.LogWarning($"Function '{_functionId}': Unable to deserialize partition or lease info with the " +
+                    $"following errors: {string.Join(" ", partitionErrors)}");
+                _nextPartitionWarningTime = DateTime.UtcNow.AddMinutes(PartitionLogIntervalInMinutes);
+            }
+
+            if (totalUnprocessedEventCount > 0 && logPartitionInfo)
+            {
+                _logger.LogInformation($"Function '{_functionId}', Total new events: {totalUnprocessedEventCount}");
+                _nextPartitionLogTime = DateTime.UtcNow.AddMinutes(PartitionLogIntervalInMinutes);
+            }
+
+            return new EventHubTriggerMetrics
+            {
+                Timestamp = DateTime.UtcNow,
+                PartitionCount = partitionRuntimeInfo.Count,
+                EventCount = totalUnprocessedEventCount
+            };
+        }
+
+        private async Task<Tuple<BlobPartitionLease, string>> GetPartitionLeaseFileAsync(int partitionId)
+        {
+            BlobPartitionLease blobPartitionLease = null;
+            string prefix = $"{EventHubOptions.GetBlobPrefix(_eventHubName, _connectionStringBuilder.Endpoint.Host)}{_consumerGroup}/{partitionId}";
+            string errorMessage = null;
+
+            try
+            {
+                CloudBlockBlob blockBlob = _blobContainer.GetBlockBlobReference(prefix);
+
+                if (blockBlob != null)
+                {
+                    var result = await blockBlob.DownloadTextAsync();
+                    if (!string.IsNullOrEmpty(result))
+                    {
+                        blobPartitionLease = JsonConvert.DeserializeObject<BlobPartitionLease>(result);
+                    }
+                }
+            }
+            catch (StorageException e)
+            {
+                if (e.RequestInformation?.HttpStatusCode == (int)HttpStatusCode.NotFound)

```

what if RequestInformation is null or it isn't a 404 - we won't log anything? We probably don't want to just swallow the exception in this case.